mtags/src/main/scala-2/scala/meta/internal/pc/SemanticdbTextDocumentProvider.scala

```diff
@@ -12,12 +12,26 @@ import scala.meta.io.AbsolutePath
 class SemanticdbTextDocumentProvider(val compiler: MetalsGlobal) {
   import compiler._
 
+  private val magicImportsRegex = """import\s+($ivy|$repo|$dep|$scalac)\..*""".r
```

Comment:
It needs to escape `$` characters.
```suggestion
  private val magicImportsRegex = """import\s+(\$ivy|\$repo|\$dep|\$scalac)\..*""".r
```

Reply:
Fixed, also move it to a separate file to also use in the Scala 3 worksheets.

Reply:
And also added a test.